src/textual/widgets/_selection_list.py

```diff
@@ -0,0 +1,655 @@
+from ._toggle_button import ToggleButton
+
+SelectionType = TypeVar("SelectionType")
+"""The type for the value of a [`Selection`][textual.widgets.selection_list.Selection] in a [`SelectionList`][textual.widgets.SelectionList]"""
+
+MessageSelectionType = TypeVar("MessageSelectionType")
+"""The type for the value of a [`Selection`][textual.widgets.selection_list.Selection] in a [`SelectionList`][textual.widgets.SelectionList] message."""
+
+
+class SelectionError(TypeError):
+    """Type of an error raised if a selection is badly-formed."""
+
+
+class Selection(Generic[SelectionType], Option):
+    """A selection for a [`SelectionList`][textual.widgets.SelectionList]."""
+
+    def __init__(
+        self,
+        prompt: TextType,
+        value: SelectionType,
+        initial_state: bool = False,
+        id: str | None = None,
+        disabled: bool = False,
+    ):
+        """Initialise the selection.
+
+        Args:
+            prompt: The prompt for the selection.
+            value: The value for the selection.
+            initial_state: The initial selected state of the selection.
+            id: The optional ID for the selection.
+            disabled: The initial enabled/disabled state. Enabled by default.
+        """
+        if isinstance(prompt, str):
+            prompt = Text.from_markup(prompt)
+        super().__init__(prompt.split()[0], id, disabled)
+        self._value: SelectionType = value
+        """The value associated with the selection."""
+        self._initial_state: bool = initial_state
+        """The initial selected state for the selection."""
+
+    @property
+    def value(self) -> SelectionType:
+        """The value for this selection."""
+        return self._value
+
+    @property
+    def initial_state(self) -> bool:
+        """The initial selected state for the selection."""
+        return self._initial_state
+
+
+class SelectionList(Generic[SelectionType], OptionList):
+    """A vertical selection list that allows making multiple selections."""
+
+    BINDINGS = [Binding("space", "select")]
+    """
+    | Key(s) | Description |
+    | :- | :- |
+    | space | Toggle the state of the highlighted selection. |
+    """
+
+    COMPONENT_CLASSES: ClassVar[set[str]] = {
+        "selection-list--button",
+        "selection-list--button-selected",
+        "selection-list--button-highlighted",
+        "selection-list--button-selected-highlighted",
+    }
+    """
+    | Class | Description |
+    | :- | :- |
+    | `selection-list--button` | Target the default button style. |
+    | `selection-list--button-selected` | Target a selected button style. |
+    | `selection-list--button-highlighted` | Target a highlighted button style. |
+    | `selection-list--button-selected-highlighted` | Target a highlighted selected button style. |
+    """
+
+    DEFAULT_CSS = """
+    SelectionList > .selection-list--button {
+        text-style: bold;
+        background: $foreground 15%;
```

Comment:
Rather than `$foreground 15%`, could we use `$boost` ?

Reply:
@willmcgugan Keep in mind that this is designed to look [exactly the same as the `Checkbox`](https://github.com/Textualize/textual/blob/8151946f380acacb8daa39b9c5b705033f5263d6/src/textual/widgets/_toggle_button.py#L70-L74) -- do you checkboxes and radio buttons changed too?

Reply:
Maybe, but check what it looks like first.

Reply:
Looking back, I think I remember why we decided to do it this way in the first place: IIRC `$foreground 15%` is the same as `$boost`, right? But there's currently no way to lighten or darken `$boost` and that was needed to allow for lifting a button when it's highlighted, darkening it when it's disabled, etc. So in the end we went with different levels of `$foreground`.

I also have a vague recollection that it didn't have the intended effect when in light mode.